### PR TITLE
Upgrade to Apache Tuweni 1.2.0

### PR DESCRIPTION
**Pull request by atoulme**
Signed-off-by: Antoine Toulme <antoine@lunar-ocean.com>

## PR description
Straight upgrade to Apache Tuweni 1.2.0. There are no API changes. Some dependencies are no longer present transitively, so I had to add them explicitly to projects.

## Changelog

- [X] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
> I didn't add to the CHANGELOG.

### CODE REVIEW

File: enclave/src/main/java/org/hyperledger/besu/enclave/EnclaveFactory.java [hunk 1]
```diff
@@ -20,10 +20,12 @@
 import java.net.URI;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.AccessDeniedException;
+import java.nio.file.Files;
```
Comment (shemnon) [inline enclave/src/main/java/org/hyperledger/besu/enclave/EnclaveFactory.java:23]: These changes are unrelated to tuweni, move them to a separate PR please.

File: ethereum/referencetests/build.gradle [hunk 2]
```diff
@@ -32,14 +32,17 @@ dependencies {
...
+  implementation 'com.google.guava:guava'
+
 
   testOutput sourceSets.test.output
 }
 
 task ('validateReferenceTestSubmodule') {
   description = "Checks that the reference tests submodule is not accidentially changed"
   doLast {
-    def expectedHash = 'b8abaebf5949877ead03e8f74ac03c0ede41d5f7'
```
Comment (shemnon) [inline ethereum/referencetests/build.gradle:42]: There is no need to update the reference tests as part of tuweni.  Please remove changes to this file and the submodule update form the PR.

File: gradle/versions.gradle [hunk 3]
```diff
@@ -31,6 +31,8 @@ dependencyManagement {
     dependency 'com.google.errorprone:error_prone_core:2.4.0'
     dependency 'com.google.errorprone:error_prone_test_helpers:2.4.0'
 
+    dependency 'com.google.code.findbugs:jsr305:3.0.2'
```
Comment (shemnon) [inline gradle/versions.gradle:34]: Is this a requirement for Tuweni?  If not please remove.

**Review by shemnon — CHANGES_REQUESTED** [review]

File: enclave/src/main/java/org/hyperledger/besu/enclave/EnclaveFactory.java (hunk 1 above)
Comment (atoulme) [reply enclave/src/main/java/org/hyperledger/besu/enclave/EnclaveFactory.java:23]: It seems this upgrade of Tuweni changes dependencies, especially removing Guava. So we have to add Guava to the dependencies, or have to use the native Files class. I chose to go down that way as it was the only reference to Guava in that particular subproject. Would you rather I change to add Guava explicitly in dependencies?

File: gradle/versions.gradle (hunk 3 above)
Comment (atoulme) [reply gradle/versions.gradle:34]: It is a requirement for Besu. It looks like those transitive dependencies are no longer there, and therefore it is required for Besu to explicitly state that dependency.

File: ethereum/referencetests/build.gradle (hunk 2 above)
Comment (atoulme) [reply ethereum/referencetests/build.gradle:42]: That must be me making a mess again. I'll fix that.

File: enclave/src/main/java/org/hyperledger/besu/enclave/EnclaveFactory.java (hunk 1 above)
Comment (shemnon) [reply enclave/src/main/java/org/hyperledger/besu/enclave/EnclaveFactory.java:23]: I would rather the dependency be changed.

File: gradle/versions.gradle (hunk 3 above)
Comment (shemnon) [reply gradle/versions.gradle:34]: It looks like it was added to the referencetests changes you are rolling back.  See if it works not adding this particular dependency once the reference tests are reset.

File: enclave/src/main/java/org/hyperledger/besu/enclave/EnclaveFactory.java (hunk 1 above)
Comment (atoulme) [reply enclave/src/main/java/org/hyperledger/besu/enclave/EnclaveFactory.java]: OK fixed.

File: gradle/versions.gradle (hunk 3 above)
Comment (atoulme) [reply gradle/versions.gradle:34]: It doesn't :(

File: ethereum/referencetests/build.gradle [hunk 4]
```diff
@@ -32,6 +32,9 @@ dependencies {
   implementation project(':services:kvstore')
 
   implementation 'com.fasterxml.jackson.core:jackson-databind'
+  implementation 'com.google.code.findbugs:jsr305'
```
Comment (shemnon) [inline ethereum/referencetests/build.gradle:35]: Looks like you can get rid of JSR305, here as well as in versions.gradle.
https://app.circleci.com/pipelines/github/hyperledger/besu/5943/workflows/f4e50074-c111-46e0-ac0c-97fdfb3036bd/jobs/28696

File: gradle/versions.gradle (hunk 3 above)
Comment (shemnon) [inline gradle/versions.gradle:34]: This can be dropped, if you also drop the dependency in `ethereum/referencetests/build.gradle`

**Review by shemnon — CHANGES_REQUESTED** [review]
* The JSR305 dependency needs to be removed.
* Merge with current

Other than those LGTM.

File: gradle/versions.gradle (hunk 3 above)
Comment (atoulme) [reply gradle/versions.gradle]: alright, done.

**Review by shemnon — APPROVED** [review]
